### PR TITLE
test_limiter_node: Build issue: GCC 4.8 bug

### PR DESCRIPTION
**Pull request by vlserov**
### Description 
Class template inheritance doesn't work well with function-local types.
Moving class definition to global scope
[GCC Commit](https://gcc.gnu.org/git/?p=gcc.git&a=commit;h=f92c74268a1f3e9ff6921097e8da418879769e2b)

Fixes # - _issue number(s) if exists_

- [x] - git commit message contains appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneTBB/blob/master/CONTRIBUTING.md#pull-requests) for details)_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add respective label(s) to PR if you have permissions_

- [x] bug fix - _change which fixes an issue_
- [ ] new feature - _change which adds functionality_
- [x] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and for some bug fixes_
- [ ] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
@anton-potapov

### CODE REVIEW

File: test/tbb/test_limiter_node.cpp [hunk 1]
```diff
@@ -605,18 +605,18 @@ TEST_CASE( "Deduction guides" ) {
 }
 #endif
 
+struct TestLargeStruct {
+    char bytes[512]{ 0 };
```
Comment (alexey-katranov) [inline test/tbb/test_limiter_node.cpp:609]: Will other element be value initialized?

Comment (alexey-katranov) [reply test/tbb/test_limiter_node.cpp:609]: Yes, see [Aggregate initialization](https://en.cppreference.com/w/cpp/language/aggregate_initialization)

> If the number of initializer clauses is less than the number of members and basesor initializer list is completely empty, the remaining members and bases are initialized by their default member initializers, if provided in the class definition, and otherwise copy-initialized from empty lists, in accordance with the usual list-initialization rules (which performs ***value-initialization for non-class types*** and non-aggregate classes with default constructors, and aggregate initialization for aggregates). If a member of a reference type is one of these remaining members, the program is ill-formed.

File: test/tbb/test_limiter_node.cpp [hunk 2]
```diff
@@ -605,18 +605,18 @@ TEST_CASE( "Deduction guides" ) {
...
 //! \brief \ref error_guessing
 TEST_CASE("Test correct node deallocation while using small_object_pool") {
-    struct TestLargeStruct {
-        char bytes[512]{ 0 };
-    };
-
     tbb::flow::graph graph;
-    tbb::flow::queue_node<TestLargeStruct> input_node{ graph };
-    tbb::flow::function_node<TestLargeStruct> func{ graph, tbb::flow::serial,
-        [](const TestLargeStruct& input) { return input; } };
```
Comment (anton-potapov) [inline test/tbb/test_limiter_node.cpp:619]: Why these changes are needed ?

Comment (vlserov) [reply test/tbb/test_limiter_node.cpp:619]: Not necessary. This is just code style change to make it look like the rest of the file.
For example, https://github.com/oneapi-src/oneTBB/blob/6ed3ba2f428cab180dd2518b155bb2fd7563e03f/test/tbb/test_limiter_node.cpp#L381

**Review by anton-potapov — APPROVED** [review]
**LGTM**
